### PR TITLE
GCE: Update Image to Ubuntu 20_04

### PR DESCRIPTION
**Pull request by dirien**
Overall issue -> #95

Signed-off-by: Engin Diri <engin.diri@mail.schwarz>

## Description
Update Ubuntu images to 20.04 for the GCE  provider

## How Has This Been Tested?

```bash
go run main.go create -p gce --project-id=$PROJECTID -f=key.json
Using provider: gce
Requesting host: fervent-turing5 in us-central1-a, from gce
2021/07/08 08:12:13 Creating firewall rule: inlets
Host: fervent-turing5|us-central1-a|central-element-274017, status: provisioning
[1/500] Host: fervent-turing5|us-central1-a|central-element-274017, status: STAGING
[2/500] Host: fervent-turing5|us-central1-a|central-element-274017, status: STAGING
[3/500] Host: fervent-turing5|us-central1-a|central-element-274017, status: STAGING
[4/500] Host: fervent-turing5|us-central1-a|central-element-274017, status: STAGING
[5/500] Host: fervent-turing5|us-central1-a|central-element-274017, status: active
inlets PRO TCP (0.8.3) server summary:
  IP: 35.224.50.33
  Auth-token: AkxuuNfu3xLOJHNURiJRnLdJx19JET13I67nZslYFkYUvPC7J55ckxzDggqznqQD
```
Create the inlets exit node:

```bash
inlets-pro tcp client --url "wss://35.224.50.33:8123" \
  --token "AkxuuNfu3xLOJHNURiJRnLdJx19JET13I67nZslYFkYUvPC7J55ckxzDggqznqQD" \
  --upstream $UPSTREAM \
  --ports $PORTS
2021/07/08 08:13:23 Starting TCP client. Version 0.8.5 - 8db64ded51b1455cf4ad027c52bc9cfbeb55c4b3
2021/07/08 08:13:23 Licensed to: Free trial <contact@openfaas.com>, expires: 29 day(s)
2021/07/08 08:13:23 Upstream server: localhost, for ports: 8080
inlets-pro client. Copyright Alex Ellis, OpenFaaS Ltd 2020
INFO[2021/07/08 08:13:24] Connecting to proxy                           url="wss://35.224.50.33:8123/connect"
INFO[2021/07/08 08:13:25] Connection established.. OK.
```
Test local service:
```bash
curl 35.224.50.33:8080
<!DOCTYPE html>
<html>
<head>
<title>Welcome to nginx!</title>
<style>
    body {
        width: 35em;
        margin: 0 auto;
        font-family: Tahoma, Verdana, Arial, sans-serif;
    }
</style>
</head>
<body>
<h1>Welcome to nginx!</h1>
<p>If you see this page, the nginx web server is successfully installed and
working. Further configuration is required.</p>

<p>For online documentation and support please refer to
<a href="http://nginx.org/">nginx.org</a>.<br/>
Commercial support is available at
<a href="http://nginx.com/">nginx.com</a>.</p>

<p><em>Thank you for using nginx.</em></p>
</body>
</html>
```
## How are existing users impacted? What migration steps/scripts do we need?
no

## Checklist:

I have:

- [x] updated the documentation and/or roadmap (if required)
- [x] read the [CONTRIBUTION](https://github.com/inlets/inlets/blob/master/CONTRIBUTING.md) guide
- [x] signed-off my commits with `git commit -s`
- [ ] added unit tests

### CODE REVIEW

File: cmd/create.go [hunk 1]
```diff
@@ -422,7 +422,7 @@ func createHost(provider, name, region, zone, projectID, userData, inletsPort st
 	} else if provider == "gce" {
 		return &provision.BasicHost{
 			Name:     name,
-			OS:       "projects/debian-cloud/global/images/debian-9-stretch-v20191121",
```
Comment (alexellis) [inline cmd/create.go:425]: I suspect there was a good reason why we used Debian here, but Stretch is now out of support.

@utsavanand2 do you remember why? cc @maelvls Perhaps Debian was just so much quicker to boot up a VM?

Comment (utsavanand2) [reply cmd/create.go:425]: @alexellis Yes, we tried out both Ubuntu and Stretch and at the time we built this, Stretch was marginally faster on the smallest f1-micro machine type. We might want to try out this minimal image and see how this works out `projects/ubuntu-os-cloud/global/images/ubuntu-minimal-2004-focal-v20210707`

Comment (dirien) [reply cmd/create.go:425]: If you want, i can change to minimal and give it a spin again

Comment (dirien) [reply cmd/create.go:425]: @alexellis or @utsavanand2, do you have any rough expected time for booting? So i can compare or should we just build on minimal and if it works its fine?

**Comment by dirien** [comment]
wit the minimal ubuntu focal image:

```bash
go run main.go create -p gce --project-id=$PROJECTID -f=key.json

Using provider: gce
Requesting host: elegant-goldwasser1 in us-central1-a, from gce
2021/07/09 07:23:43 Creating firewall exists, updating: inlets
Host: elegant-goldwasser1|us-central1-a|central-element-274017, status: provisioning
[1/500] Host: elegant-goldwasser1|us-central1-a|central-element-274017, status: STAGING
[2/500] Host: elegant-goldwasser1|us-central1-a|central-element-274017, status: STAGING
[3/500] Host: elegant-goldwasser1|us-central1-a|central-element-274017, status: STAGING
[4/500] Host: elegant-goldwasser1|us-central1-a|central-element-274017, status: active
inlets PRO TCP (0.8.3) server summary:
  IP: 35.192.104.197
  Auth-token: rsYwL1OjqT5tNOOw12em6ixXoNooxk6z9AfrJXuRrlJyqmw5HHMFLjXJB9iMKKG1

inlets-pro tcp client --url "wss://35.192.104.197:8123" \
  --token "rsYwL1OjqT5tNOOw12em6ixXoNooxk6z9AfrJXuRrlJyqmw5HHMFLjXJB9iMKKG1" \
  --upstream $UPSTREAM \
  --ports $PORTS
```

```bash
2021/07/09 07:24:54 Starting TCP client. Version 0.8.5 - 8db64ded51b1455cf4ad027c52bc9cfbeb55c4b3
2021/07/09 07:24:54 Licensed to: Free trial <contact@openfaas.com>, expires: 28 day(s)
2021/07/09 07:24:54 Upstream server: localhost, for ports: 8080
inlets-pro client. Copyright Alex Ellis, OpenFaaS Ltd 2020
INFO[2021/07/09 07:24:56] Connecting to proxy                           url="wss://35.192.104.197:8123/connect"
INFO[2021/07/09 07:24:56] Connection established.. OK.
```


```bash
 ~ curl 35.192.104.197:8080
<!DOCTYPE html>
<html>
<head>
<title>Welcome to nginx!</title>
<style>
    body {
        width: 35em;
        margin: 0 auto;
        font-family: Tahoma, Verdana, Arial, sans-serif;
    }
</style>
</head>
<body>
<h1>Welcome to nginx!</h1>
<p>If you see this page, the nginx web server is successfully installed and
working. Further configuration is required.</p>

<p>For online documentation and support please refer to
<a href="http://nginx.org/">nginx.org</a>.<br/>
Commercial support is available at
<a href="http://nginx.com/">nginx.com</a>.</p>

<p><em>Thank you for using nginx.</em></p>
</body>
</html>
```

**Comment by alexellis** [comment]
Merging. That seems fast enough 👍 

Thanks both

**Pull request merged**